lib/src/implicitly_animated_reorderable_list.dart

```diff
@@ -491,20 +445,49 @@ class ImplicitlyAnimatedReorderableListState<E>
     });
   }
 
-  void onDragEnded() {
-    if (dragKey == null || _closestList.isEmpty) return;
+  ///
+  /// Finding destination of drag item.
+  /// Items can move to up/down so position can be changed while index is not change.
+  /// We need virtual index reflecting moves.
+  /// Empty space of virtual indexes for non-drag item will be destination index of drag item.
+  ///
+
+  _Item findSwapTargetItem() {
```

Comment:
```suggestion
  _Item findSwapTargetItem() {
    // Account for the fact, that a move could have been scheduled,
    // without the widget having fully rebuild itself. In that case, the translation
    // would be 0.0 (and thus the index calculation wrong) and controller.isAnimating will be true.
    //
    // This can be the case, when the user released his finger just
    // short moments after a move had been scheduled (using _dispatchMove()).
    for (final item in _itemBoxes.values) {
      if (item == dragItem) continue;

      final translation = getTranslation(item.key);
      final controller = _itemTranslations[item.key];
      if (translation == 0.0 && controller?.isAnimating == true) {
        return item;
      }
    }
```
Sometimes, when you release an item just after another one had been scheduled to move (up or down), the dragged item would still go to the wrong place, because the translation of the item is still 0.0. By inserting that snippet before, this should not happen.

Reply:
I confirmed problem and resolving, thanks!